doc/nrf/app_dev/device_guides/nrf54l/kmu_provision.rst

```diff
@@ -56,62 +59,85 @@ If the SoC has been previously provisioned and you need to use a different set o
 
-The script generates the public key for each private key and uploads them to your device.
-These public keys generate the verification keys for the application image, which are then used by MCUboot for validation.
-The first key specified in the command is used for signing the application image.
-Currently, the script supports only ED25519 Keys.
+            - keyname: UROT_PUBKEY
+                keys: ["/path/private-key1.pem", "/path/private-key2.pem"]
+                policy: lock
+            - keyname: APP_PUBKEY
+                keys: ["/path/private-key3.pem", "/path/private-key4.pem"]
+                policy: lock
 
-For MCUboot, take note of the following:
+      * Parameter ``--dry-run`` specifies that a command should generate a keyfile for nRF Util without actually executing the command.
 
-* By default, it uses one key.
-* KMU support in its configuration needs to be enabled by setting the ``SB_CONFIG_MCUBOOT_SIGNATURE_USING_KMU`` sysbuild Kconfig option.
-  Otherwise, MCUboot will fallback to the compiled in key.
+      The script generates the public key for each private key and uploads them to your device.
+      These public keys generate the verification keys for the application image, which are then used by MCUboot for validation.
+      The first key specified in the command is used for signing the application image.
+      Currently, the script supports only ED25519 Keys.
 
-For provision one key to the board run the following command:
+      For MCUboot, take note of the following:
 
-.. parsed-literal::
-   :class: highlight
+      * By default, it uses one key.
+      * KMU support in its configuration needs to be enabled by setting the ``SB_CONFIG_MCUBOOT_SIGNATURE_USING_KMU`` sysbuild Kconfig option.
+        Otherwise, MCUboot will fallback to the compiled in key.
+
+      For provision one key to the board run the following command:
+
+      .. parsed-literal::
+        :class: highlight
+
+          west ncs-provision upload -s nrf54l15 -k ed25519.pem --keyname UROT_PUBKEY
+
+   .. tab:: nRF Util
+
+      The nRF Util provisioning command requires a JSON file with the keys and the key metadata.
+
+      You can use the `generate_psa_key_attributes.py`_ script, :ref:`similarly to nRF54H20<ug_nrf54h20_keys_generating>`, to generate the JSON file and the metadata from the PEM file you :ref:`generated earlier <ug_nrf54l_developing_provision_kmu_generate>`.
+      For this purpose, invoke the script with the ``--key-from-file`` option to provide the PEM file and with the ``--file`` option to create the JSON file.
+      The file can contain multiple keys: calling the script multiple times and pass the same file to the `--file` argument will add all keys to the JSON file.
+
+      To provision keys onto the KMU of the nRF54L15 SoC, use the following nRF Util command, with the ``<snr>`` being the serial number of the device and ``<key-file>`` being the name of the key file in the JSON format:
```

Comment:
Where the key is stored is based on information in the metadata string, currently it's two locations, one for each device family supported:

PERSISTENT_CRACEN (nrf54h) or PERSISTENT_CRACEN_KMU (nrf54l)

These are options given to the  --location argument. I am not sure if or how you can get this information into the document.

Reply:
This should be covered in the (currently missing) script documentation ([task](https://nordicsemi.atlassian.net/browse/NCSIDB-1497)).